boa_engine/src/syntax/ast/node/yield/mod.rs

```diff
@@ -29,13 +29,12 @@ impl Yield {
     }
 
     /// Creates a `Yield` AST node.
-    pub fn new<E, OE>(expr: OE, delegate: bool) -> Self
+    pub fn new<E>(expr: Option<E>, delegate: bool) -> Self
```

Comment:
The reason behind this was to allow any `E` to be used, since `T` implements `Into<Option<T>>`, so we could use the type directly without wrapping it in a `Some`. On the other hand, this increases compile time, and the benefit might not even be apparent, since for `None` you might need to turbo fish.

I guess that's the reasoning behind removing it, right?

Reply:
Yeah, also if the turbofish would only be needed for `None` that would be fine, but for some reason we also need it in some cases for `Some`. For example here: https://github.com/boa-dev/boa/blob/56be7cec523c091e6cc49419542f985cbf344106/boa_engine/src/syntax/parser/expression/assignment/yield.rs#L72-L75

That double type specification seems really unnecessary to me, especially because the function is only used in so few places.